plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,104 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '20em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const catalogApi = useApi(catalogApiRef);
```

Comment:
I think I'm going to go ahead and suggest something unorthodox here. You see what this does to tests, and probably also will have led to subtle breakages in storybooks too, and people out there will have to suffer equally in rewriting their tests when they upgrade to a version that has this released.

I think you should do `useApiHolder()` instead and try to get the catalog client out of that. And if the catalog client isn't supplied, it just renders nothing. Either disabling the popup functionality entirely, or showing the popup still but only with a small admonition saying "Could not fetch details from the catalog." or whatnot.

Then you can revert all of the test changes in the rest of the repo.